pages/hints/rendering-lists.md

```diff
@@ -0,0 +1,135 @@
+experiencing any delay. The visual issue presents as a duplicate `a` cat image until
+image `b` is downloaded, at which point it replaces the first `a` image.
+
+To try and represent this in a different way:
+
+```
+Increment ->                 Increment ->
+             <img href="a">  --changes-->  <img href="b">
+                             -- added -->  <img href="a">
+```
+
+> **Note:** Because the cat API returns a random image, it is possible for the final
+> result to be a duplicate cat image. This is a different situation from the one
+> described above. Using the slugs from above, this situation would produce
+> a list `[a, a]`.
+
+### Fixing the quickstart program
+
+As mentioned in the first section above, the way to fix this issue is to [key the
+elements](https://hexdocs.pm/lustre/lustre/element.html#keyed). This allows Lustre
+to better track which items have changed, moved, been removed, etc. Keyed elements
+were intentionally not used in the quickstart tutorial in an attempt to focus on
+the fundamentals.
+
+The key must be unique within a list, and the cat API could return the same image
+since it's random. This means the image slug is not suitable for use as a key.
+One solution is to make a synthetic key and pass it around as part of the model.
+This means we don't have to add additional depenencies such as a UUID library.
+
+> **Note:** Lustre will display a warning in the browser console when it detects
+> duplicate keys.
+
+Update the model so it carries the next synthetic id. The list of cats now also includes
+an `id` along with the URL slug, which will be used as the element key:
+
+```gleam
+pub type Model {
+  Model(count: Int, cats: List(#(Int, String)), next_id: Int)
+}
+```
+
+The `init` function needs updating for the new model constructor:
+
+```gleam
+fn init(_flags) -> #(Model, effect.Effect(Msg)) {
+  #(Model(0, [], 0), effect.none())
+}
+```
+
+In the `update` function, during the `ApiReturnedCat(Ok(cat))` message match, the
+`next_id` is used when adding the new cat to the list. The `next_id` is also 
+incremented so any subsequent uses of it will have a new value. The 
+`UserDecrementedCount` message match needs to be updated to support the new model
+constructor also.
+
+```gleam
+pub fn update(model: Model, msg: Msg) -> #(Model, effect.Effect(Msg)) {
+  case msg {
+    UserIncrementedCount -> #(Model(..model, count: model.count + 1), get_cat())
+    UserDecrementedCount -> #(
+      Model(model.count - 1, list.drop(model.cats, 1), model.next_id),
+      effect.none(),
+    )
+    ApiReturnedCat(Ok(cat)) -> #(
+      Model(
+        model.count,
+        [#(model.next_id, cat), ..model.cats],
+        model.next_id + 1,
+      ),
+      effect.none(),
+    )
+    ApiReturnedCat(Error(_)) -> #(model, effect.none())
+  }
+}
+```
+
+Finally, update the `view` function to use `element.keyed`:
+
+```gleam
+pub fn view(model: Model) -> element.Element(Msg) {
```

Comment:
I'd like this bit to be the only code snippet, and ideally not even a complete `view` function but instead specifically focused on the use of `element.keyed`.

The purpose of these "hints" pages is to provide concise information on a niche or tricky thing rather than be a follow-along guide, so lets make sure we're higlighting the "thesis" of this page and removing any of the setup.

Reply:
Hopefully addressed in commit [fddb549](https://github.com/lustre-labs/lustre/pull/186/commits/fddb5491572d7dfcf18044e91f0cb64df22c9001)